imap_processing/tests/mag/test_mag_decom.py

```diff
@@ -60,15 +59,16 @@ def test_mag_raw_xarray():
     assert burst_data.dims["epoch"] == expected_burst_len
 
 
-@pytest.mark.xfail(reason="Fix for updated cdflib")
 def test_mag_raw_cdf_generation(tmp_path):
     current_directory = Path(__file__).parent
     test_file = current_directory / "mag_l0_test_data.pkts"
     l0 = decom_packets(str(test_file))
 
+    output_path = tmp_path / "imap" / "mag" / "l1a" / "2023" / "10"
+
     norm_data, burst_data = export_to_xarray(l0)
 
-    test_data_path_norm = tmp_path / "mag_l1a_raw-normal_20210101_20210102_v01-01.cdf"
+    test_data_path_norm = output_path / "imap_mag_l1a_norm-raw_20231025_v001.cdf"
 
     assert not test_data_path_norm.exists()
     output = write_cdf(norm_data, test_data_path_norm)
```

Comment:
You should check this, but I was thinking that this might actually not be testing what we think it is here. I think it might be creating a path with that filename as a part of the directory structure `imap_mag_l1a_norm-raw_20231025_v001.cdf/imap/mag/...`

In the other tests I just removed all of the `tmp_path` fixtures and asserted based on t he filename from `output.name == expected_name`, which might simplify this a bit. Then we can rely on the `write_cdf()` tests for the specific directory structure expectations.

Reply:
Ok, I think I understand what you mean. I'll update this now.